### DISCUSSION TITLE
CS Tools Google Sheets Syncer Error

### DISCUSSION THREAD

**Discussion by leonliflipp** [thoughtspot/cs_tools#110] [Q&A]
Hi,

I was trying to use the Google Sheets syncer on CS tools but I'm running into an issue. 
The command I'm running is: `cs_tools tools searchable bi-server --syncer gsheets:///$HOME/Documents/TS_CS_Tools/google-sheets-definition.toml --compact`. 

My definition.toml file is:
[configuration]
spreadsheet = 'test'
mode = 'overwrite'
credentials_file = '/Users/leonli/Documents/TS_CS_Tools/google-sheets-creds.json'

This is a screenshot of the error.
![image](https://github.com/thoughtspot/cs_tools/assets/141184264/d170498c-662b-4847-8bb3-0f66274b28b6)

And this is the log report:
[2024-03-12T10_35_29.log](https://github.com/thoughtspot/cs_tools/files/14574370/2024-03-12T10_35_29.log)

Thanks!

**Comment by leonliflipp** [comment]
I also have some additional questions:

1. Since we are specifying one google sheet in the definition file, what's the expected behaviour when `searchable gather` is ran? Does it dump all the metadata into one google sheet?
2. Does the google sheet just have to live somewhere on my drive or e.g. can it be in a folder?

Thanks again!

**Reply by boonhapus** [reply]
1. The gsheet is the file itself in your Google Drive. It'll dump metadata into separate tabs in the sheet. Google confusingly calls these _both_ "spreadsheets". 🫠
2. I believe it can live anywhere. I believe the scoping happens based on sharing the file to your service account user. I am not sure how Google behaves if you've shared 2 google sheets with the same name.

![image](https://github.com/thoughtspot/cs_tools/assets/14192437/339ec87c-1d37-49e6-a6b1-ddae2d69da44)

**Comment by boonhapus** [comment]
Hey @leonliflipp , can you switch the order of `mode` and `credentials_file` ? This should be fixed in 1.5.0 .

```toml
[configuration]
spreadsheet = 'ThoughtSpot Data Sink'
credentials_file = 'C:\Users\boonhapus\Downloads\syncers\cs-tools-user-sync-02cf6ee39e60.json'
mode = 'overwrite'
```

**Reply by leonliflipp** [reply]
Hi @boonhapus,

I switched it but it still gives me the same error. 

Here's the log error when I switched it (if needed):
[2024-03-12T11_38_26.log](https://github.com/thoughtspot/cs_tools/files/14575097/2024-03-12T11_38_26.log)

**Reply by boonhapus** [reply]
Hmmmm okay, try one last thing @leonliflipp ... remove `mode` altogether (its default is to overwrite).

**Reply by leonliflipp** [reply]
@boonhapus Still doesn't work 😔 (giving same error)

**Reply by boonhapus** [reply]
Gotcha. I confirmed this does work in the upcoming new version, to be released in the next couple of weeks. Do you need this functionality now? Or can you wait until the new version is released?

**Reply by leonliflipp** [reply]
I can wait until the new version is released. Does next couple of weeks mean around the end of the month? Also when it is released, can you let me know? Thank you again for looking into this, much appreciated!!

**Reply by boonhapus** [reply]
Yep! Exactly. My goal is before April to finish documentation and release it. Will set a reminder to poke you in Slack as well. Thanks @leonliflipp !
